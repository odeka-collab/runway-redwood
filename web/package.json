{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults"
    ]
  },
  "dependencies": {
    "@phosphor-icons/react": "^2.0.10",
    "@redwoodjs/forms": "6.0.6",
    "@redwoodjs/router": "6.0.6",
    "@redwoodjs/web": "6.0.6",
    "animejs": "^3.2.1",
    "prop-types": "15.8.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "uuid": "^9.0.0",
    "zdog": "^1.1.3",
    "zfont": "^1.2.8"
  },
  "devDependencies": {
    "@redwoodjs/vite": "6.0.6",
    "autoprefixer": "^10.4.14",
    "postcss": "^8.4.27",
    "postcss-loader": "^7.3.3",
    "tailwindcss": "^3.3.3"
  }
}
