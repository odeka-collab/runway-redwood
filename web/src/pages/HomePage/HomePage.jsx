import { MetaTags } from '@redwoodjs/web'

import RunwayWizard from 'src/components/RunwayWizard/RunwayWizard'

const HomePage = () => {
  return (
    <>
      <MetaTags title="Runway App" description="How much runway do you have?" />
      <RunwayWizard />
    </>
  )
}

export default HomePage
