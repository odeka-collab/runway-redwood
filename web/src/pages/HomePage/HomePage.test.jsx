import { render } from '@redwoodjs/testing/web'

import HomePage from 'src/pages/HomePage'
import Providers from 'src/providers'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('HomePage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(
        <Providers>
          <HomePage />
        </Providers>
      )
    }).not.toThrow()
  })

  it('renders the onboarding form', () => {
    expect(
      render(
        <Providers>
          <HomePage />
        </Providers>
      )
    ).toMatchSnapshot()
  })
})
