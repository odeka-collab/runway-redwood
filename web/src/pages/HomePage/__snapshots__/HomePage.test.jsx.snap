// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders the onboarding form 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <div
          class="flex flex-col sm:gap-8"
        >
          <h2
            class="py-4 text-center text-3xl"
          >
            Howdy!
          </h2>
          <p
            class="text-md mx-auto px-8 py-2 text-center text-lg sm:w-7/12"
          >
            This tool is built for the harsh financial realities we are all dealing with. Check out a sample financial runway, or try it for yourself!
          </p>
          <p
            class="mx-auto px-6 py-2 text-center text-base sm:w-2/3"
          >
            Sh*t just got real. No one's financial future is certain. Look at a few scenarios we've already been through.
          </p>
          <div
            class="grid content-stretch gap-6 p-6 sm:grid-cols-2"
          >
            <button
              class="h-36 rounded-3xl border-4 border-double border-black p-4 text-xl xs:p-8 sm:h-auto sm:py-24"
            >
              Living off the business with friends
            </button>
            <button
              class="h-36 rounded-3xl border-4 border-double border-black p-4 text-xl xs:p-8 sm:h-auto sm:py-24"
            >
              I was just laid off
            </button>
          </div>
          <div
            class="flex flex-col justify-between gap-6 p-6 pt-0 sm:flex-row"
          >
            <button
              class="rounded-lg border-4 border-double border-black px-4 py-2 uppercase"
            >
              <span
                class="flex items-center justify-center gap-2"
              >
                Try it myself
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M237.66,122.34l-96-96A8,8,0,0,0,128,32V72H48A16,16,0,0,0,32,88v80a16,16,0,0,0,16,16h80v40a8,8,0,0,0,13.66,5.66l96-96A8,8,0,0,0,237.66,122.34ZM144,204.69V176a8,8,0,0,0-8-8H48V88h88a8,8,0,0,0,8-8V51.31L220.69,128Z"
                  />
                </svg>
              </span>
            </button>
            <button
              class="rounded-lg border-4 border-double border-black px-4 py-2 uppercase"
            >
              <span
                class="flex items-center justify-center gap-2"
              >
                I know what I'm doing
                <svg
                  class="h-6 w-auto"
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M152,88a32,32,0,1,0-32-32A32,32,0,0,0,152,88Zm0-48a16,16,0,1,1-16,16A16,16,0,0,1,152,40Zm67.31,100.68c-.61.28-7.49,3.28-19.67,3.28-13.85,0-34.55-3.88-60.69-20a169.31,169.31,0,0,1-15.41,32.34,104.29,104.29,0,0,1,31.31,15.81C173.92,186.65,184,207.35,184,232a8,8,0,0,1-16,0c0-41.7-34.69-56.71-54.14-61.85-.55.7-1.12,1.41-1.69,2.1-19.64,23.8-44.25,36.18-71.63,36.18A92.29,92.29,0,0,1,31.2,208,8,8,0,0,1,32.8,192c25.92,2.58,48.47-7.49,67-30,12.49-15.14,21-33.61,25.25-47C86.13,92.35,61.27,111.63,61,111.84A8,8,0,1,1,51,99.36c1.5-1.2,37.22-29,89.51,6.57,45.47,30.91,71.93,20.31,72.18,20.19a8,8,0,1,1,6.63,14.56Z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <div
        class="flex flex-col sm:gap-8"
      >
        <h2
          class="py-4 text-center text-3xl"
        >
          Howdy!
        </h2>
        <p
          class="text-md mx-auto px-8 py-2 text-center text-lg sm:w-7/12"
        >
          This tool is built for the harsh financial realities we are all dealing with. Check out a sample financial runway, or try it for yourself!
        </p>
        <p
          class="mx-auto px-6 py-2 text-center text-base sm:w-2/3"
        >
          Sh*t just got real. No one's financial future is certain. Look at a few scenarios we've already been through.
        </p>
        <div
          class="grid content-stretch gap-6 p-6 sm:grid-cols-2"
        >
          <button
            class="h-36 rounded-3xl border-4 border-double border-black p-4 text-xl xs:p-8 sm:h-auto sm:py-24"
          >
            Living off the business with friends
          </button>
          <button
            class="h-36 rounded-3xl border-4 border-double border-black p-4 text-xl xs:p-8 sm:h-auto sm:py-24"
          >
            I was just laid off
          </button>
        </div>
        <div
          class="flex flex-col justify-between gap-6 p-6 pt-0 sm:flex-row"
        >
          <button
            class="rounded-lg border-4 border-double border-black px-4 py-2 uppercase"
          >
            <span
              class="flex items-center justify-center gap-2"
            >
              Try it myself
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M237.66,122.34l-96-96A8,8,0,0,0,128,32V72H48A16,16,0,0,0,32,88v80a16,16,0,0,0,16,16h80v40a8,8,0,0,0,13.66,5.66l96-96A8,8,0,0,0,237.66,122.34ZM144,204.69V176a8,8,0,0,0-8-8H48V88h88a8,8,0,0,0,8-8V51.31L220.69,128Z"
                />
              </svg>
            </span>
          </button>
          <button
            class="rounded-lg border-4 border-double border-black px-4 py-2 uppercase"
          >
            <span
              class="flex items-center justify-center gap-2"
            >
              I know what I'm doing
              <svg
                class="h-6 w-auto"
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M152,88a32,32,0,1,0-32-32A32,32,0,0,0,152,88Zm0-48a16,16,0,1,1-16,16A16,16,0,0,1,152,40Zm67.31,100.68c-.61.28-7.49,3.28-19.67,3.28-13.85,0-34.55-3.88-60.69-20a169.31,169.31,0,0,1-15.41,32.34,104.29,104.29,0,0,1,31.31,15.81C173.92,186.65,184,207.35,184,232a8,8,0,0,1-16,0c0-41.7-34.69-56.71-54.14-61.85-.55.7-1.12,1.41-1.69,2.1-19.64,23.8-44.25,36.18-71.63,36.18A92.29,92.29,0,0,1,31.2,208,8,8,0,0,1,32.8,192c25.92,2.58,48.47-7.49,67-30,12.49-15.14,21-33.61,25.25-47C86.13,92.35,61.27,111.63,61,111.84A8,8,0,1,1,51,99.36c1.5-1.2,37.22-29,89.51,6.57,45.47,30.91,71.93,20.31,72.18,20.19a8,8,0,1,1,6.63,14.56Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
